CHANGELOG.md

```diff
@@ -44,6 +44,7 @@ _ `_optional` subpackage for managing optional dependencies
   `SubspaceContinuous.sample_uniform`
 - `SubspaceContinuous.samples_full_factorial` has been replaced with
   `SubspaceContinuous.sample_from_full_factorial`
+- `add_fake_results` and `add_parameter_noise` have been moved to the simulation package
```

Comment:
These were my thoughts:
* In my eyes, both these methods are inherently related to "simulation", i.e. doing stuff that does NOT happen in the real world <-- that's my definition of simulation (instead of just interpreting "simulation" as "replay"). The motivation was thus: "You need some functionality that invents/fantasizes stuff? -> You'll find it in the simulation package"
* In fact, it was very weird to me that these two functions were sitting in `utils` while very similar things such as "lookup" are in the simulation package. After all, they conceptually do the same thing: fill in / create values
* One could argue that these function are not pure "utilities" like the rest that's in the utilities package, because they are couple to baybe objects and implement some higher-level logic. The remaining stuff that you find in "utils" is really just pure python utils that do low-level stuff.

If you both disagree with the above points, I'm open for debate. But then let me hear your thoughts against my arguments :D

Reply:
I think your first point is wrong, Ive already given an example where we are not in the simulation setting (the loop-like backtesting or measurements that our main functions there offer) and want to use these fantasies

Reply:
But how is this not "simulation" then? In this case, you are very well "simulating" (or call it fantasizing, if you prefer) values. These values are not real, they don't come from actual measurement. Don't see the argument, tbh?

Reply:
you have come up with a definition of `simulation` never voiced before, so far for me simulation was `loop for backtest or measurements via lookup`

Reply:
Ok, fair enough. But don't you think it's inconsistent that "lookup" stuff in a different place than these to functions? After all, can you tell me single situation where you apply one but not the other? I think the set of possible use cases is **exactly** the same for all functions!?

Reply:
again my initial example: a jupyer nb where you create some recommendations manually, perturb them , add random data etc, loop this

no lookup anywhere

Reply:
Yes, so your "loop-closing step" is "invent numbers", which can be 1:1 replaced with functionality that – for some reason, however – sits in a different package. So let me try with these last two remarks. If they don't convince you, I give up 😄:
* You'd prefer to do `from babye.utils.dataframe import add_fake_results`, but you get *the exact same thing* if you do `from baybe.simulation.lookup import look_up_targets` and apply that function instead, which really just calls your lower-level routine. In fact, that said, one could even argue for making the two helpers private.
* The high-level `look_up_targets` clearly shows: wherever you can apply `add_fake_results`, you can ALSO apply other mechanism, i.e. their set of use cases matches 1:1. Weirdly, though, they sit in different places.

So even if you disagree with the move, I'd like to hear your thoughts on the above. If we don't come to agreement here, I'll revert the move, since it just blocks us from moving forward.

Reply:
not surprising that the usecases overlap strongly and certainly no argument against the split. After all `lookup_targets` uses the **utility** for filling the dataframe.

The same argument holds for:

1) to model a substance, I cans use a custom parameter with our chemical descriptors imported from `utils.chem`
2) However, the same is possible with `SubstanceParameter`
Wherever I want to use 1) I can also use 2)

weirdly enough, they sit in different packages?

Reply:
Alright, as I said, let me then revert the changes so the we can move on. But a this really opens up a related question that we **should** resolve: 

I can follow your argument with them `chem` and `SubstanceParameter` and agree with the logic here. However, I think we both agree the for substances, the `SubstanceParameter` way is clearly the recommended one, as it provides the high-level user interface. 

Now you can (and should, if you say that these two situations are analogous) apply the exact same reasoning to `add_fake_results` and `look_up_targets`. Either we say that the latter is the "user-facing" high-level dispatcher, but then we should also use this in our examples. Everything else would be similar to exposing a `SubstanceParameter` but then using the custom-encoding approach in all user examples. Or, we say that the fake-data part of `look_up_targets` is not so obvious and can potentially even lead to unintended user setups (<-- I think this is my opinion, tbh), but then we should also remove this case.

Reply:
@Scienfitz: last open thread before merge.

Reply:
you seem to expect a reply, but I would simply resolve this?

Reply:
Yes, I do expect a reply 😄 Namely to the last paragraph. In my eyes, we should accordingly delete the "None" option from the lookup utility

Reply:
I don't understand the requirement that a utility has to be exclusively either for its independent usage OR in a high-level dispatcher... why not both?

A user might as well want to generate molecule fingerprints (not with a substance parameter but for whatever it might be), why should he not be able to use our utility for that? And how is this even connected to what SubstanceParameter does? Analogously for `lookup_targets`, no change required imo

Reply:
My point is: if using `look_up_targets` without a lookup (i.e. producing fake data) is useful in any way, how come that we don't ever use it, not even in all our examples where exactly that operation is done. If not even we use it, who else would? Thus my conclusion was: all that having this subroutine part of the dispatcher brings is i) more complicated signature of the dispatcher ii) more complicated docstring.

If you want to keep it, I will not block it. But if not even you are using it, I think it will pretty much appear nowhere in any real code...

Reply:
I don't see strong reason to remove it in the dispatcher but if its important top you then do it
perhaps let @AVHopp decide

Reply:
Not sure if I was able to follow all of your arguments, but let me try to share my opinion.

First of all, why do we have separate user-facing functions having the same functionailty? There should not be an `add_fake_results` (or whatever it is called) if this is equivalent to using a lookup with `None`. This would also make the question of where to move that function redundant as it would no eixst. For the parameter noise: I think this was not really discussed here, right? Bit for simplicity, I would keep it where it is. But I think we should discuss this point in our next dev meeting since there seem to be some strong opinions here.

Reply:
And I do not have a strong opinion on where to put this stuff, I see points for having this in `utils` and some for having it in `simulation`, I do not think that there is a general answer here, we just need to decide for something. My gut feeling/opinion (if you desperately need a decision) would be to keep this stuff in `utils`.

Reply:
To explain the current situation `add_fake_results` is the implementation of the specific case in `look_up_targets` when no lookup is provided. `look_up_targets` is merely a dispatched that calls the lower level routines depending on the given input. So the question is not so much if we want to remove either of the two, but perhaps rather if we want to expose both possibilities as public entry points to the user... In order to move forward, I'll keep stuff as is for now, but perhaps we can collect the next steps for the `simulation` package in general in our next dev meeting, because finishing the refactoring is really overdue...